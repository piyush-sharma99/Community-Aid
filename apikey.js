/*
 *  ClassName: apikey.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://reactnative.dev/docs/activityindicator
 * @reference https://www.udemy.com/course/react-native-the-practical-guide/
 * @reference https://www.youtube.com/watch?v=8zT6CYu0iYQ
 *
 */

export const firebaseConfig = {
  apiKey: "AIzaSyDGliQfMfKpXUllG5ZPyClJL6C8VgFKXo8",
  authDomain: "community-aid.firebaseapp.com",
  databaseURL:
    "https://community-aid-default-rtdb.europe-west1.firebasedatabase.app",
  projectId: "community-aid",
  storageBucket: "community-aid.appspot.com",
  messagingSenderId: "529896701339",
  appId: "1:529896701339:web:1c30369d39f8654b01353e",
  measurementId: "G-PPD28P0PC6",
};
