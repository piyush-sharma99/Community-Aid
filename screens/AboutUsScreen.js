/*
 *  ClassName: AppAssistanceSceen.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://reactnative.dev/docs/activityindicator
 *
 */

//Imports
import React from "react";
import { StyleSheet, Text, View, ImageBackground } from "react-native";

//Main Component
const AboutUsScreen = (props) => {
  return (
    <ImageBackground source={require("../assets/BG.png")} style={styles.bImage}>
      <View style={styles.screen}>
        <View>
          <Text style={styles.text}>
            Community Aid is a community based mobile application that can be
            used by users to ask for assistance when needed. The targets
            audience of the application is individuals of all ages above 18.
            This is because the purpose of this application is to help
            individuals gain assistance when needed. My main personal
            inspiration for developing this application was a family member who
            is an elderly woman that has been stuck in her house during the time
            of Covid-19. I help her by doing her weekly shopping. After a few
            conversations with her I realized that it is exceedingly difficult
            for individuals in her position to receive help. Therfore, come on
            and help make the world a better place one assistance request at a
            time.
          </Text>
        </View>
      </View>
    </ImageBackground>
  );
};

//Navigation options: changing header displayed on the page
AboutUsScreen.navigationOptions = {
  headerTitle: "About us",
  headerStyle: {
    backgroundColor: "#2c8ffa",
  },
  headerTintColor: "white",
};

//CSS
const styles = StyleSheet.create({
  screen: {
    width: "100%",
    height: "100%",
    flex: 1,
  },
  bImage: {
    width: "100%",
    height: "100%",
  },
  text: {
    padding: 10,
    marginRight: 40,
    marginTop: 30,
    marginLeft: 40,
    color: "white",
    fontSize: 20,
    textAlign: "center",
    backgroundColor: "#fb5b5a",
    borderWidth: 3,
    borderColor: "white",
    shadowOpacity: 2,
    elevation: 5,
    shadowOffset: {
      width: 3,
      height: 3,
    },
  },
});

export default AboutUsScreen;
