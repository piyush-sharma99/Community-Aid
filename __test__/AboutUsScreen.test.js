import React from "react";
import AboutUsScreen from "../screens/AboutUsScreen";
import { firebaseConfig } from "../apikey";
import { cleanup } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);

afterEach(cleanup);

test("AboutUs renders correctly", () => {
  const tree = renderer.create(<AboutUsScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
