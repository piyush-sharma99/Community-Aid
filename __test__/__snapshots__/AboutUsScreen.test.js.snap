// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "backgroundColor": "#fb5b5a",
            "borderColor": "white",
            "borderWidth": 3,
            "color": "white",
            "elevation": 5,
            "fontSize": 20,
            "marginLeft": 40,
            "marginRight": 40,
            "marginTop": 30,
            "padding": 10,
            "shadowOffset": Object {
              "height": 3,
              "width": 3,
            },
            "shadowOpacity": 2,
            "textAlign": "center",
          }
        }
      >
        Community Aid is a community based mobile application that can be used by users to ask for assistance when needed. The targets audience of the application is individuals of all ages above 18. This is because the purpose of this application is to help individuals gain assistance when needed. My main personal inspiration for developing this application was a family member who is an elderly woman that has been stuck in her house during the time of Covid-19. I help her by doing her weekly shopping. After a few conversations with her I realized that it is exceedingly difficult for individuals in her position to receive help. Therfore, come on and help make the world a better place one assistance request at a time.
      </Text>
    </View>
  </View>
</View>
`;
